cmd/jaeger/internal/exporters/storageexporter/exporter.go

```diff
@@ -47,5 +52,5 @@ func (*storageExporter) close(_ context.Context) error {
 }
 
 func (exp *storageExporter) pushTraces(ctx context.Context, td ptrace.Traces) error {
-	return exp.traceWriter.WriteTraces(ctx, td)
+	return exp.traceWriter.WriteTraces(ctx, exp.sanitizer(td))
```

Comment:
in exporter_test we write a trace with empty resource attributes. Then trace is retrieved from memory storage and verified. You can add a check to that verification confirming that service name attribute was added by the sanitizer.